### DISCUSSION TITLE
how to use params dict in a "get" request

### DISCUSSION THREAD

**Discussion by jonathanbouchet** [fastapi/fastapi#10583] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Path

app = FastAPI()

items = [{"foo": "bar"}, {"john": "doe"}]


@app.get("/item/{item_id}",
         description="get item by id",
         tags=["test"])
async def read_item(item_id: int = Path(..., description="the id of the item to get")):
    return {"item": items[item_id]}
```

### Description

Hi,
I'm still new to FastAPI so apologies in advance if my question is naive.
Let's say I have a get endpoint (please see code) to retrieve a dictionary from a list.

I've been trying to use this endpoint in a request by doing:

```
import requests
import json
url = "http://127.0.0.1:8000/item"
params = {"item_id":0}
r=requests.get(url, params=params)
```

I tried to add the header as well (`headers = {"accept":"application/json","Content-Type":"application/json"}`) but it doesn't work.
The swaggerUI gives the request URL though (`http://127.0.0.1:8000/item/0`) which works but I was wondering if a `params` dictionary can be used instead.

thanks in advance

### Operating System

macOS

### FastAPI Version

0.104.0

### Pydantic Version

2.4.2

### Python Version

Python 3.11.4

**Comment by jgould22** [comment, marked as answer]
When you write this 
```
params = {"item_id":0}
r=requests.get(url, params=params)
```

requests is going to create a request to `http://127.0.0.1:8000/item?item_id=0` 

but you have told FastAPI that you want item id from the path by putting it in the brackets in the path
```python
@app.get("/item/{item_id}",...
```

 which is a request like this `http://127.0.0.1:8000/item/0`

If you want the endpoint to work like your example request you need to use query params https://fastapi.tiangolo.com/tutorial/query-params/#query-parameters

**Reply by jonathanbouchet** [reply]
thanks a lot @jgould22 

I used a `Query` parameter in the endpoint as you suggested

```
@app.get("/item_from_request/",
         description="get item by id",
         tags=["test"])
async def read_item(item_id: int = Query(..., description="the id of the item to get")):
    return {"item": items[item_id]}
```

and I'm able to do the following now:
```
params =  {"item_id":0}
requests.get("http://127.0.0.1:8000/item_from_request/", params=params).json()
#  {'item': {'foo': 'bar'}}
```
which was my goal.
